designs/virtualization/4.6/Next-run-configuration/Next-run-configuration.md

```diff
@@ -0,0 +1,62 @@
+---
+parent: Virtualization
+version: 4.6
+---
+
+# Editing
+
+When a user clicks to edit/ update one of the attributes (f.e. Flavor) it will open a modal with an alert notification letting the user know that the VM must be restarted in order to see the changes applied. The alert will persist until they restart the VM.
+Alternatively they could clear the changes and keep what they had.
+To avoid confusion they will be able to view the current settings in the VM Details page.
+
+![user is editing values](img/EditOriginal.png)
+
+At this point the user can either just save the requested changes and postpone the restart or save and restart the VM.
+
+## Restart modal
+
+If the user chooses to save and restart the VM they will be asked again to ensure that and will get a modal including an inline notification that restarting may be disruptive to some logged in users and a notification that the new setting will be applied upon restart. The notification will include a confirmation checkbox so they would be aware of the effect that the restart action might cause and a link to the logged in users list at the bottom of the VM Details page.
+
+![user chooses to resart the VM](img/RestartModal.png)
+
+## VM Details page
+
+In the VM Details page they will be notified that the changes they have made will get applied only after the VM will be restarted. 
+Next to the ‘Flavor’ section we will add a link that will lead to the Flavor modal where they can view the pending changes.
+Only in cases when users have made changes to the part of the YAML which are not exposed in the ui, we will add a link to ‘View pending changes in YAML editor’.
+
+![VM details page](img/vm-details.png)
+
+In cases users made changes to the part of the YAML which are not exposed in the ui we’ll add the option to ‘View pending changes in the YAML editor.
+
+![Users can view the changes in the YAML](img/ViewYAML-editor.png)
+
+## List view
+
+In the List view, we’ll show the Running* status. Clicking it will open a popover explaining that this VM has some pending changes that will apply once it will be restarted.
```

Comment:
Thanks @lizsurette  Did you mean add an alert icon to the status column instead of the *?

Reply:
Exactly. Curious what others think of the idea too :) @andybraren @matthewcarleton

Reply:
Thanks. I'll add both options so we can see better.

Reply:
ya I commented on the comment added by Andy. I think we could step back and look at the list view now that we are adding another "*". It might be out of scope of this effort but I like the idea of aligning with what we currently do and add the icon. In the future it could be nice to have an official status for Next Run and missing GA and drop the "*"

Reply:
By "alert icon" are we talking about the yellow warning triangle in the current design? If the inline alert within the Details page and modals changes to become a blue info type instead, a warning icon in the List may not align with what's shown there. This is a tricky balance. 😄

An option that might be more clear than the `*` could be `Changes pending` underneath the regular `Running` status. Kind of like what we do for operators:

![image](https://user-images.githubusercontent.com/9122899/82356698-f2dce980-99d1-11ea-9210-f6082690b41f.png)

Reply:
ya another reason why this is probably out of scope for this effort. I think there's room to improve the status column in general. 
@andybraren this is from the status card though right not in a list view? Be interesting to see how the list view would look with this. I'm all for some "Just in time" help :)

Reply:
I agree it can definitely be clearer @andybraren but I wonder how it would look when not all the rows in the same height. I will create a mock-up and share it.

Reply:
It's from a list view yeah, sorry, I should've zoomed out. 🙂

![image](https://user-images.githubusercontent.com/9122899/82402652-f781bc00-9a2a-11ea-9fad-7898111c33c0.png)

Reply:
I really like the idea of using the subtext!

Reply:
@andybraren @matthewcarleton @lizsurette WDYT of the 3rd option for the status suggested in the list view screen?

Reply:
Thinking more, one big benefit of treating un-restarted VMs as "degraded" with yellow warning statuses in List page is that they would/should appear in the Cluster Overview and (IIRC) Project Overview Inventory cards as well. That should make it even easier to notice VMs that need to be restarted. So yeah, good point @yfrimanm, sounds good to me! 👍 Should we mention that in this documentation?

And the substatus text lgtm!
![image](https://user-images.githubusercontent.com/9122899/82464045-9807c880-9a8b-11ea-8b3d-aff73e2797b8.png)

Reply:
nice! Ya I think we can create a follow up for the GA stuff to update the list view there too.

Reply:
This looks excellent to me!

Reply:
Thank you all @andybraren @matthewcarleton @lizsurette! Documentation & mockup are updated.